agreementbot/agreementworker.go

```diff
@@ -216,12 +216,37 @@ func (b *BaseAgreementWorker) InitiateNewAgreement(cph ConsumerProtocolHandler,
 			return
 		}
 
-		// check if workload is suspended
 		if wi.ConsumerPolicy.PatternId != "" {
+			// check if workload is suspended. if suspended, then do not make agreement
 			if found, suspended := exchange.ServiceSuspended(exchangeDev.RegisteredServices, workload.WorkloadURL, workload.Org); found && suspended {
 				glog.Infof(BAWlogstring(workerId, fmt.Sprintf("cannot make agreement with %v for policy %v because service %v is suspended by the user.", wi.Device.Id, wi.ConsumerPolicy.Header.Name, cutil.FormOrgSpecUrl(workload.WorkloadURL, workload.Org))))
 				return
 			}
+
+			// check the arch of the top level service against the node. If not match, then do not make agreement.
+			// In the pattern case, the top leverl service spec, arch and version are also put in the node's registeredServices
```

Comment:
Typo here "leverl" -> "level"
Also, I really wish this piece of code was in it's own function instead of just inline like this.

Reply:
Thanks for the review. I changed the typo part, not the I-wish part,